src/main/java/com/example/cinematicketbookingapp/mapper/ReservationDtoMapper.java

```diff
@@ -0,0 +1,54 @@
+package com.example.cinematicketbookingapp.mapper;
+
+import com.example.cinematicketbookingapp.config.AppFunctionalValuesConstants;
+import com.example.cinematicketbookingapp.dto.ReservationCreationDataDto;
+import com.example.cinematicketbookingapp.dto.ReservationSummaryDto;
+import com.example.cinematicketbookingapp.exceptions.ResourceNotFoundException;
+import com.example.cinematicketbookingapp.model.Reservation;
+import com.example.cinematicketbookingapp.model.Screening;
+import com.example.cinematicketbookingapp.model.Seat;
+import com.example.cinematicketbookingapp.repository.ScreeningRepository;
+import com.example.cinematicketbookingapp.repository.SeatRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class ReservationDtoMapper {
+
+    private final ScreeningRepository screeningRepository;
+    private final SeatRepository seatRepository;
+
+    public Reservation mapToReservation(ReservationCreationDataDto reservationCreationDataDto) {
+        return Reservation.builder()
+                .screening(getScreeningById(reservationCreationDataDto.screeningId()))
+                .ownerName(reservationCreationDataDto.ownerName())
+                .ownerSurname(reservationCreationDataDto.ownerSurname())
+                .creationDateTime(LocalDateTime.now())
+                .seats(getSeatsById(reservationCreationDataDto.seatIds()))
+                .build();
+    }
+
+    public ReservationSummaryDto mapToReservationSummaryDto(Reservation reservation) {
+        return ReservationSummaryDto.builder()
+                .reservationId(reservation.getId())
+                .expirationTime(reservation.getCreationDateTime().plus(AppFunctionalValuesConstants.RESERVATION_EXPIRATION_TIME))
+                .build();
+    }
+
+    private Screening getScreeningById(Long id) {
+        return screeningRepository.findById(id)
+                .orElseThrow(() -> new ResourceNotFoundException("Screening", id));
+    }
+
+    private Set<Seat> getSeatsById(Set<Long> seatIds) {
+        return seatIds.stream()
+                .map(seatId -> seatRepository.findById(seatId)
```

Comment:
To nie jest najbardziej wydajny sposób. Dlaczego i jak to naprawić?

Reply:
No tak, zapycham bazę danych wieloma pojedynczymi zapytaniami, które baza musie przetworzyć, zamiast jednym porządnym i w sumie byłem tego świadomy, ale spojrzałem na sytuację z perspektywy biznesowej i uznałem, że przecież przypadki, w których będzie zajmowanych na raz kilka miejsc (np. więcej niż 5) będą bardzo rzadko występowały. Żeby to naprawić, można by było zrobić pojedyncze zapytanie wykorzystując słowo kluczowe IN np.: 
`@Query("SELECT seat FROM Seat seat WHERE seat.id IN :ids")`
`Set<Seat> findAllSeatWhenIdContainedInIdSet(Set<Long> ids);`. 
Takie rozwiązanie wykorzystałby z moją aktualną wiedzą, ale wygooglowałem jaki jest najwydajniejszy sposób na pobieranie danych z bazy danych za pomocą słowa kluczowego `IN` i gdyby kluczowym aspektem była wydajność oraz liczba encji `Seat` była by bardzo duża to skorzystałbym z metody, która polegałaby na wstawieniu tymczasowej tabeli do bazy danych z kolumną `ID(long)` z wartościami odpowiadającymi ID’kom z `Set<Long> seatIds` a następnie zrobieniu `SELECT z INNER JOIN` tej tymczasowej tabeli z tabelą `Seats` po `ID`. W sumie ten drugi sposób nie wykorzystuje słowa kluczowego `IN`, ale jest ponoć najwydajniejszym substytutem.

Reply:
Myślę że IN spokojnie by wystarczył.